calypso/contracts.go

```diff
@@ -51,6 +55,9 @@ func (c *contractWr) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instructi
 			err = errors.New("couldn't unmarshal write: " + err.Error())
 			return
 		}
+		if d := inst.Spawn.Args.Search("darcID"); d != nil {
```

Comment:
Could you explain what this part is used for? Does it means the client who spawns the contract can choose any darc in byzcoin to be the darc that protects the write request? I'm curious whether this approach will have security implications (probably not coz the one that spawns the write request has full responsibility of who gets to read it). Also, if this is the case, can we have a test for it?

Reply:
Yes, this means that the client who spawns a write request can define which darc will protect the read requests. It is used when spawning a WriteInstance from a SpawnerInstance, or if you want to create a WriteInstance from a Darc, but want to give another darc for the control.

For security implications, you can 'hijack' a darc and its access patterns, but besides giving other people access to your WriteInstance, there is nothing more I can think of.

With regard to tests, this is currently only tested in TypeScript, but not in go.

Reply:
Sounds good. See my response regarding tests in the other comment.